packages/explorerkit-server/src/components/idls.ts

```diff
@@ -0,0 +1,63 @@
+import { SolanaFMParser } from "@solanafm/explorer-kit";
+import { getProgramIdl, IdlItem } from "@solanafm/explorer-kit-idls";
+
+import { getSharedDep } from "@/core/shared-dependencies";
+
+export type IdlsMap = Map<string, SolanaFMParser | null>;
+
+const IDL_CACHE_TTL = 3600; // one hour
+
+export async function loadAllIdls(programIds: string[]): Promise<IdlsMap> {
+  const idls: IdlsMap = new Map();
+  const cache = getSharedDep("cache");
+  const cachedIdls = await Promise.allSettled(programIds.map((id) => cache.get(id)));
+
+  await Promise.allSettled(
```

Comment:
Could we add some observability around cache hits and misses? We lost some observability when moving to redis but I think hits & misses covers the most important things

Reply:
Yeah, sure. Instrumented cache client here: https://github.com/blowfishxyz/explorer-kit/pull/4/files#diff-d3009e49bb0e9aeba73f3f1c4908e24128f36d4b2cd05d1bb488e711c3f411c0R22-R58